test/sql/join/test_nested_payloads.test

```diff
@@ -0,0 +1,330 @@
+# name: test/sql/join/test_nested_payloads.test
+# description: Test join with nested types in the payload
+# group: [join]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE integers(fk INTEGER)
+
+statement ok
+INSERT INTO integers VALUES (1), (2), (3), (5), (NULL)
+
+# Payload is LIST<INTEGER>
+statement ok
+CREATE VIEW intlists AS SELECT * FROM (VALUES
+	(1, [1]),
+	(2, [NULL]),
+	(3, []),
+	(4, [2, 3]),
+	(5, [9,10,11]),
+	(NULL::INTEGER, [13])
+	) lv(pk, p);
+
+query II
+SELECT pk, p FROM intlists ORDER BY pk;
+----
+NULL	[13]
+1	[1]
+2	[NULL]
+3	[]
+4	[2, 3]
+5	[9, 10, 11]
+
+query III
+SELECT fk, pk, p FROM integers, intlists WHERE fk = pk ORDER BY fk, pk
+----
+1	1	[1]
+2	2	[NULL]
+3	3	[]
+5	5	[9, 10, 11]
+
+query III
+SELECT fk, pk, p FROM integers LEFT OUTER JOIN intlists ON integers.fk=intlists.pk ORDER BY fk, pk
+----
+NULL	NULL	NULL
+1	1	[1]
+2	2	[NULL]
+3	3	[]
+5	5	[9, 10, 11]
+
+query III
+SELECT fk, pk, p
+FROM integers LEFT OUTER JOIN intlists ON integers.fk=intlists.pk
+WHERE pk < 5 AND fk > 1
+ORDER BY fk, pk
+----
+2	2	[NULL]
+3	3	[]
+
+query III
+SELECT fk, pk, p FROM intlists RIGHT OUTER JOIN integers ON integers.fk=intlists.pk ORDER BY fk, pk
+----
+NULL	NULL	NULL
+1	1	[1]
+2	2	[NULL]
+3	3	[]
+5	5	[9, 10, 11]
+
+query III
+SELECT fk, pk, p FROM integers FULL OUTER JOIN intlists ON integers.fk=intlists.pk ORDER BY fk, pk
+----
+NULL	NULL	NULL
+NULL	NULL	[13]
+NULL	4	[2, 3]
+1	1	[1]
+2	2	[NULL]
+3	3	[]
+5	5	[9, 10, 11]
+
+# Payload is LIST<VARCHAR>
+statement ok
+CREATE VIEW strlists AS SELECT * FROM (VALUES
+	(1, ['a']),
+	(2, [NULL]),
+	(3, []),
+	(4, ['b', 'c']),
+	(5, ['i','j','k']),
+	(NULL::INTEGER, ['m'])
+	) lv(pk, p);
+
+query III
+SELECT fk, pk, p FROM integers, strlists WHERE fk = pk ORDER BY fk
+----
+1	1	[a]
+2	2	[NULL]
+3	3	[]
+5	5	[i, j, k]
+
+query III
+SELECT fk, pk, p FROM integers LEFT OUTER JOIN strlists ON integers.fk=strlists.pk ORDER BY fk
+----
+NULL	NULL	NULL
+1	1	[a]
+2	2	[NULL]
+3	3	[]
+5	5	[i, j, k]
+
+query III
+SELECT fk, pk, p
+FROM integers LEFT OUTER JOIN strlists ON integers.fk=strlists.pk
+WHERE pk < 5 AND fk > 1
+ORDER BY fk
+----
+2	2	[NULL]
+3	3	[]
+
+query III
+SELECT fk, pk, p FROM strlists RIGHT OUTER JOIN integers ON integers.fk=strlists.pk ORDER BY fk
+----
+NULL	NULL	NULL
+1	1	[a]
+2	2	[NULL]
+3	3	[]
+5	5	[i, j, k]
+
+query III
+SELECT fk, pk, p FROM integers FULL OUTER JOIN strlists ON integers.fk=strlists.pk ORDER BY fk, pk
+----
+NULL	NULL	NULL
+NULL	NULL	[m]
+NULL	4	[b, c]
+1	1	[a]
+2	2	[NULL]
+3	3	[]
+5	5	[i, j, k]
+
+# Payload is STRUCT<a: INTEGER, b: VARCHAR>
+statement ok
+CREATE VIEW structs AS SELECT * FROM (VALUES
+	(1, {'a': 1, 'b': 'a'}),
```

Comment:
Could you add a test with longer strings? These strings are all `<=12` bytes and will thus be inlined which won't test the string heap changes correctly.